testing/sync-test/src/auth.rs

```diff
@@ -237,40 +50,20 @@ pub struct TestClient {
 }
 
 impl TestClient {
-    pub fn new(acct: Arc<TestAccount>) -> Result<Self> {
-        log::info!("Doing oauth flow!");
-        let mut fxa = FirefoxAccount::with_config(acct.cfg.clone());
-        // We either authenticate using the normal oauth_flow
-        // Or we use a pairing flow with a logged in account
-        // Both should work fine in executing the oauth flow
-        let (code, state) = if rand::random() {
-            let pairing_url = acct.cfg.authorization_endpoint().unwrap();
-            let pairing_url = fxa.begin_pairing_flow(
-                pairing_url.as_str(),
-                &[SYNC_SCOPE],
-                "integration_test",
-                None,
-            )?;
-            acct.execute_oauth_pair_flow(&pairing_url)?
-        } else {
-            let oauth_uri = fxa.begin_oauth_flow(&[SYNC_SCOPE], "integration_test", None)?;
-            let redirect_uri = acct.execute_oauth_flow(&oauth_uri)?;
-            let redirect_uri = Url::parse(&redirect_uri)?;
-            let query_params = redirect_uri
-                .query_pairs()
-                .into_owned()
-                .collect::<HashMap<String, String>>();
-            (query_params["code"].clone(), query_params["state"].clone())
+    pub fn new(acct: Arc<Account>, device_name: &str) -> Result<Self> {
+        // XXX - not clear if/how this device gets cleaned up!
```

Comment:
The device is only destroyed when we disconnect the account (calling `disconnect`)